pkg/providers/provider.go

```diff
@@ -34,6 +35,9 @@ type Provider interface {
 	CleanUp(bool, *v2vv1.VirtualMachineImport, rclient.Client) error
 	FindTemplate() (*oapiv1.Template, error)
 	ProcessTemplate(*oapiv1.Template, *string, string) (*kubevirtv1.VirtualMachine, error)
+	NeedsGuestConversion() bool
+	GetGuestConversionJob() (*batchv1.Job, error)
```

Comment:
Is the guest conversion job tailored only for Vmware? Can it be used as is for other providers? If it is universal, maybe it should be implemented outside of this package and called from the controller?

Reply:
In the near future, it won't be used for other providers. The next provider will likely be OpenStack and it doesn't require a conversion as it is based on KVM.

Reply:
:+1:

Reply:
Since the managers are owned by the provider the creation of the configmap and job resources needs to be done there, but the portion that generates the job spec and the libvirt domain xml can be extracted into a shared location.